packages/ui-components/src/ExpandableComponent.tsx

```diff
@@ -57,112 +57,88 @@ export interface IExpandableComponentProps {
   onMouseDown?: (e: React.MouseEvent<HTMLSpanElement, MouseEvent>) => any;
 }
 
-export interface IExpandableComponentState {
-  expanded: boolean;
-}
-
 /**
  * A React component for expandable containers.
  */
-export class ExpandableComponent extends React.Component<
-  IExpandableComponentProps,
-  IExpandableComponentState
-> {
-  constructor(props: IExpandableComponentProps) {
-    super(props);
-    this.state = { expanded: false };
-  }
+export const ExpandableComponent: React.FC<IExpandableComponentProps> = props => {
+  const [expanded, setExpandedValue] = React.useState(false);
 
-  toggleDetailsDisplay(): void {
+  const toggleDetailsDisplay = (): void => {
     // Switch expanded flag
-    const newExpandFlag = !this.state.expanded;
-    if (this.props.onBeforeExpand) {
-      this.props.onBeforeExpand(newExpandFlag);
+    const newExpandFlag = !expanded;
+    if (props.onBeforeExpand) {
+      props.onBeforeExpand(newExpandFlag);
     }
-    this.setState({ expanded: newExpandFlag });
-  }
+    setExpandedValue(newExpandFlag);
+  };
 
-  componentDidUpdate(): void {
-    if (this.props.onExpand) {
-      this.props.onExpand(this.state.expanded);
-    }
-  }
+  React.useEffect((): void => {
+    props.onExpand && props.onExpand(expanded);
+  });
```

Comment:
we shouldn't need an effect for this, this should be able to go in `handleToggleDetailsDisplay`

Reply:
Code snippets needs this callback triggered only after the component state changed to expanded, so that it renders the snippet preview. Calling this inside `handleToggleDetailsDisplay` messes up the rendering order, the first time expanding a code snippet nothing is displayed.

Reply:
oh gotcha!